### DISCUSSION TITLE
how to prevent creation of empty %{name}-%{version}-build directory when running rpmbuild -bs

### DISCUSSION THREAD

**Discussion by decathorpe** [rpm-software-management/rpm#3381] [General]
I noticed after upgrading to Fedora 41 that very common operations like `rpmbuild -bs` or `fedpkg srpm` (which calls `rpmbuild -bs` under the hood) now create an empty directory of the form `%{name}-%{version}-build`. It looks like this is part of the "hermetic build directory" stuff that was implemented in RPM 4.20.

Why does this directory get created when just assembling an SRPM too? It just creates a lot of clutter (empty directories everywhere), and messes up shell completion for tools like `mock`.

I grepped through /usr/lib/rpm/ but I can't find any way to override this behaviour with a macro that I could set in my `~/.rpmmacros` file. Can it be turned off somehow? Or should it be disabled for the `rpmbuild -bs` operation? It doesn't seem useful at all there.

**Comment by pmatilai** [comment]
Most likely that's just an oversight, aka a bug.

**Comment by pmatilai** [comment]
Filed as https://github.com/rpm-software-management/rpm/issues/3382

**Comment by decathorpe** [comment]
Thanks! I wasn't able to tell whether it's an intentional change or not, so I filed a discussion instead of a bug. :)
